### PR TITLE
feat: Adding Github Unittests workflow

### PR DESCRIPTION
**Pull request by lucifercr07**

### CODE REVIEW

**Review by Patterbear — COMMENTED** [review]
I know this is the github action way of doing unit tests, but I originally wrote all the tests with pytest and I feel it was less complicated and more readable, and also because it allows for a variety of tests,.

I personally prefer a more modular, purely functional testing approach, however if @rothoma2 feels unittest and this current format you've used should be the standard, then that's all good and i'll write future tests this way.

Anyway, the actions and the tests look pretty good.

**Comment by Patterbear** [comment]
Also, are force pushes ok as its lucifer07's own branch, or should we discourage them to set a standard when we have multiple people working on one branch? @rothoma2

**Comment by rothoma2** [comment]
Im not very familiar woth force pushes and multiple people on the same branch.

I think normally each one will have its own branch and you will merge one into master and then merge master into branch two before creating the second pullrequest.

In between you can always merge branch A into Branch2 if u need to pull changes into your branch.

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        python-version: '3.11'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint poetry
+        poetry install
+
+    - name: Run tests
+      run: poetry run python -m unittest discover -s tests/ -p '*.py'
```
Comment (rothoma2) [inline .github/workflows/unit-tests.yml:26]: I think you could do this just with pytest instead of unitest

**Review by rothoma2 — CHANGES_REQUESTED** [review]

**Comment by rothoma2** [comment]
@lucifercr07  Will you change this back to use pytest, or should we drop this PullRequest? Making sure it doesnt linger for too long.

**Pull request merged**